tempodb/backend/raw.go

```diff
@@ -134,7 +137,12 @@ func (r *reader) Blocks(ctx context.Context, tenantID string) ([]uuid.UUID, erro
 }
 
 func (r *reader) BlockMeta(ctx context.Context, blockID uuid.UUID, tenantID string) (*BlockMeta, error) {
-	bytes, err := r.r.Read(ctx, MetaName, KeyPathForBlock(blockID, tenantID))
+	reader, size, err := r.r.Read(ctx, MetaName, KeyPathForBlock(blockID, tenantID), false)
+	if err != nil {
+		return nil, err
+	}
+
+	bytes, err := tempo_io.ReadAllWithEstimate(reader, size)
```

Comment:
what happens if size is a negative value?  i think we'd panic.

can any of the backends return negative?

Reply:
👍    handled in `ReadAllWithEstimate()`